script.js

```diff
@@ -1,12 +1,121 @@
-//You can edit ALL of the code here
+// level 350
+let allEpisodes; // global scope to be accessible from inside all functions
+
+function fetchAllEpisodes() {
+  return fetch("https://api.tvmaze.com/shows/82/episodes")
+    .then((response) => response.json())
+    .then((data) => {
+      allEpisodes = data; // this was done to move allEpisodes to global scope
+      makePageForEpisodes(allEpisodes);
+    })
+    .catch(function (error) {
+      console.log(error);
+    });
+}
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  fetchAllEpisodes();
+  // makePageForEpisodes(allEpisodes);
 }
 
 function makePageForEpisodes(episodeList) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  //rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  // level 100
+  document.getElementById("root").innerHTML = "";
+  for (i = 0; i < episodeList.length; i++) {
+    // step 1 - create elements
+    let episodeDivContainer = document.createElement("div");
+    let episodeTitleElement = document.createElement("h3");
+    let episodeImageElement = document.createElement("img");
+    let episodeBlurbElement = document.createElement("p");
+
+    // assign the divs classes
+    episodeDivContainer.classList.add("episode-container");
+    episodeTitleElement.classList.add("episode-title");
+    episodeImageElement.classList.add("episode-image");
+    episodeBlurbElement.classList.add("episode-blurb");
+
+    // assign unique ID to individual episode-container
+    episodeDivContainer.id = "episode-container" + i;
+
+    // step 2 - contents of elements
+    episodeTitleElement.innerHTML = `${episodeList[i].name}-S${String(
+      episodeList[i].season
+    ).padStart(2, "0")}E${String(episodeList[i].number).padStart(2, "0")}`;
+    episodeImageElement.src = `${episodeList[i].image.medium}`;
+    episodeBlurbElement.innerHTML = `${episodeList[i].summary}`;
+
```

Comment:
i found this part of your code hard to follow. i had to google why you were using padStart and what it does.this was a was a great learning experience for me so great work.